app/views/charges/downgrade.html.erb

```diff
@@ -0,0 +1,4 @@
+<% if current_user.has_role? :premium %>
+    <h4>Click to downgrade to Standard user!</h4>
+    <%= link_to "Downgrade", method: :downgrade, class: 'btn btn-danger', data: { confirm: 'Are you sure you want to downgrade to standard user and relinquish power?' } %>
```

Comment:
@praveenyr this line is invoking the downgrade method directly, no view needed. Since you are using current_user method in the controller, no arguments need to be passed.